zachsinterview/src/main/java/com/imprev/zachsinterview/NumericalKeyValueFunImpl.java

```diff
@@ -0,0 +1,35 @@
+package com.imprev.zachsinterview;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class NumericalKeyValueFunImpl implements NumericalKeyValueFun {
+    
+    private NumericalKeyValueStore keyValueStore;
+    
+    public NumericalKeyValueFunImpl(NumericalKeyValueStore keyValueStore) {
+        this.keyValueStore = keyValueStore;
+    }
+
+    @Override
+    public Integer sumByPrefix(String prefix) {
+        
+        int sum = 0;
+        try {
+            Iterator<String> iter = this.keyValueStore.getKeysByPrefix(prefix);
+            List<String> allKeys = new ArrayList<String>();
```

Comment:
You don't need to put the keys in the list. You can use the keys directly from the iterator.